src/asm.rs

```diff
@@ -273,9 +275,9 @@ pub fn dump_range(
 
             empty_line = false;
             #[allow(clippy::match_bool)]
-            match fmt.full_name {
-                true => safeprintln!("{line:#}"),
-                false => safeprintln!("{line}"),
+            match fmt.name_display {
+                crate::opts::NameDisplay::Full => safeprintln!("{line:#}"),
```

Comment:
Somewhere around here the decision is made to how to print the line, 
This how a sample fragment looks with `--full-name`
```
        mov rsi, r13
        call <alloc::boxed::Box<T,A> as core::ops::drop::Drop>::drop::h48380e3ba0521fa1
        mov rdi, r14
        call <alloc::boxed::Box<T,A> as core::ops::drop::Drop>::drop::h06bf8e07228f6c4f
        mov rdi, rbx
        call _Unwind_Resume@PLT
        ud2
```
Same fragment, but without any options
```
        mov rsi, r13
        call <alloc::boxed::Box<T,A> as core::ops::drop::Drop>::drop
        mov rdi, r14
        call <alloc::boxed::Box<T,A> as core::ops::drop::Drop>::drop
        mov rdi, rbx
        call _Unwind_Resume@PLT
        ud2
```

I was kind of expecting for `--keep-mangled` to display mangled name, but it displays the short name instead. Is it something you are interested in implementing? I suspect some global variables shenanigans are going to be needed...

Either that or clarify in the help message that it only displays unmangled names in the selection menu.

Reply:
Hey, yeah, my intention was to demangle the names only in selection menu, though - come to think of it - it'd make sense to demangle them for labels too (in the same vein of "unmangled names can be useful too"). I'll give that a try today.

Reply:
Implemented in #227